src/components/Integrations/IntegrationsTable.tsx

```diff
@@ -0,0 +1,199 @@
+  IntegrationsTableProps
+> = (props) => {
+  const intl = useIntl();
+
+  const onSort = React.useCallback(
+    (event, column: number, direction: SortByDirection) => {
+      const propsOnSort = props.onSort;
+      const mapping = sortMapper.find((p) => p.index === column);
+      if (propsOnSort && mapping) {
+        propsOnSort(
+          mapping.index,
+          mapping.name,
+          direction === SortByDirection.asc
+            ? Direction.ASCENDING
+            : Direction.DESCENDING
+        );
+      }
+    },
+    [props.onSort]
+  );
+
+  const sortBy = React.useMemo<ISortBy>(() => {
+    const propsSortBy = props.sortBy;
+    if (propsSortBy) {
+      const mapping = sortMapper.find((p) => p.name === propsSortBy.column);
+      if (mapping) {
+        return {
+          index: mapping.index,
+          direction:
+            propsSortBy.direction === Direction.ASCENDING
+              ? SortByDirection.asc
+              : SortByDirection.desc,
+        };
+      }
+    }
+    return {
+      defaultDirection: SortByDirection.asc,
+    };
+  }, [props.sortBy]);
+
+  const rows = React.useMemo(() => {
+    return props.integrations.map((integration, idx) => [
+      integration.name,
+      Config.integrations.types[integration.type].name,
+      integration.lastConnectionAttempts === undefined ? (
+        <StatusUnknown />
+      ) : (
+        <IntegrationStatus
+          status={integration.status}
+          lastConnectionAttempts={
+            integration.isConnectionAttemptLoading
+              ? undefined
+              : integration.lastConnectionAttempts
+          }
+          includeDetails={integration.includeDetails}
+        />
+      ),
+      integration.isEnabledLoading ? (
+        <Spinner className="pf-v5-u-ml-sm" size="md" />
+      ) : (
+        <Switch
+          id={`table-row-switch-id-${integration.id}`}
+          aria-label="Enabled"
+          isChecked={integration.isEnabled}
+          onChange={(_e, isChecked) =>
+            props.onEnable && props.onEnable(integration, idx, isChecked)
+          }
+          isDisabled={!props.onEnable}
+          ouiaId={`enabled-${integration.id}`}
+        />
+      ),
+    ]);
+  }, [props.integrations, props.onEnable]);
+
+  const COLUMNS: DataViewTh[] = [
+    {
+      cell: 'Name',
+      props: { sort: { sortBy: sortBy, onSort: onSort, columnIndex: 1 } },
+    },
+    'Type',
+    'Last connection attempt',
+    {
+      cell: 'Enabled',
+      props: { sort: { sortBy: sortBy, onSort: onSort, columnIndex: 4 } },
+    },
+  ];
+
+  if (props.isLoading) {
+    return (
+      <div
+        {...getOuiaProps('Integrations/Table', { ...props, ouiaSafe: false })}
+      >
+        <SkeletonTable
+          rows={
+            props.loadingCount && props.loadingCount > 0
+              ? props.loadingCount
+              : 10
+          }
+          columns={COLUMNS}
+        />
+      </div>
+    );
+  }
```

Comment:
This can be improved https://react-data-view-pr-data-view-61.surge.sh/extensions/data-view/components#loading-state-example